chalice/deploy/planner.py

```diff
@@ -908,6 +916,14 @@ def _plan_restapi(self, resource):
             ] + shared_plan_epilogue
         else:
             deployed = self._remote_state.resource_deployed_values(resource)
+            if deployed.get('endpoint_type',
```

Comment:
working through this at the moment, its going to involve doing a couple of redundant api calls (exists calls get, but changing that to return non boolean is non trivial / invasive) which i can leave for a followup related to #1177, so we'll end up doing at least two api calls non conditionally, one redundantly. testing out of the behavior on apigw for trying a no-op patch, else this would mean growing some additional conditional evaluation onto api method.